iceoryx2/tests/sample_tests.rs

```diff
@@ -0,0 +1,165 @@
+        publisher_2: Publisher<Sut, u64>,
+        subscriber: Subscriber<Sut, u64>,
+    }
+
+    impl<Sut: Service> TestFixture<Sut> {
+        fn new() -> Self {
+            let service_name = generate_name();
+            let service = Sut::new(&service_name)
+                .publish_subscribe()
+                .max_publishers(2)
+                .max_subscribers(1)
+                .create::<u64>()
+                .unwrap();
+
+            let publisher_1 = service.publisher().create().unwrap();
+
+            let publisher_2 = service.publisher().create().unwrap();
+
+            let subscriber = service.subscriber().create().unwrap();
+
+            Self {
+                service_name,
+                service,
+                publisher_1,
+                publisher_2,
+                subscriber,
+            }
+        }
+    }
+
+    #[test]
+    fn origin_is_tracked_correctly<Sut: Service>() {
+        let test = TestFixture::<Sut>::new();
+
+        assert_that!(test.publisher_1.send_copy(123), eq Ok(1));
+        let sample = test.subscriber.receive().unwrap().unwrap();
+        assert_that!(sample.origin(), eq test.publisher_1.id());
+
+        assert_that!(test.publisher_2.send_copy(123), eq Ok(1));
+        let sample = test.subscriber.receive().unwrap().unwrap();
+        assert_that!(sample.origin(), eq test.publisher_2.id());
+    }
+
+    #[test]
+    fn sample_of_dropped_service_does_not_block_new_service_creation<Sut: Service>() {
+        let test = TestFixture::<Sut>::new();
+
+        let service_name = test.service_name;
+
+        assert_that!(test.publisher_1.send_copy(5), eq Ok(1));
+        let sample = test.subscriber.receive().unwrap();
+        assert_that!(sample, is_some);
+
+        drop(test);
+
+        assert_that!(
+            Sut::new(&service_name).publish_subscribe().create::<u64>(),
+            is_ok
+        );
+    }
+
+    #[test]
+    fn when_everything_is_dropped_the_sample_can_still_be_consumed<Sut: Service>() {
+        let test = TestFixture::<Sut>::new();
+
+        let sut = test.service;
+        let publisher_1 = test.publisher_1;
+        let publisher_2 = test.publisher_2;
+        let subscriber = test.subscriber;
+
+        drop(sut);
+
+        const PAYLOAD: u64 = 8761238679123;
+
+        assert_that!(publisher_1.send_copy(PAYLOAD), eq Ok(1));
+        let sample = subscriber.receive().unwrap().unwrap();
+
+        drop(subscriber);
+        drop(publisher_1);
+        drop(publisher_2);
+
+        assert_that!(*sample, eq PAYLOAD);
+    }
+
+    #[test]
+    fn sample_received_from_dropped_publisher_does_not_block_new_publishers<Sut: Service>() {
+        let test = TestFixture::<Sut>::new();
+
+        let publisher = test.publisher_1;
+
+        assert_that!(publisher.send_copy(123554), eq Ok(1));
+        let sample = test.subscriber.receive().unwrap().unwrap();
+
+        drop(publisher);
+
+        const PAYLOAD: u64 = 123981235645;
```

Comment:
If there are other changes and you need to create new commits, can you have a `PAYLOAD_1 = 123554` and a `PAYLOAD_2 = 1239...`

Similar in the following test.